src/MessageHandler/DatasetSubmissionFilerHandler.php

```diff
@@ -165,7 +169,10 @@ private function processFile(File $file, array $loggingContext): void
         $file->setStatus(File::FILE_IN_PROGRESS);
 
         try {
-            $newFileDestination = $this->datastore->addFile(['fileStream' => fopen($filepath, 'r')]);
+            $newFileDestination = $this->datastore->addFile(
+                ['fileStream' => fopen($filepath, 'r')],
+                str_replace(':', '.', $loggingContext['udi']) . DIRECTORY_SEPARATOR . $file->getFileName()
```

Comment:
The UDI retrieved from the logging context now has significantly more weight than when it was just a logging variable.
Maybe here we should pass in the UDI in the function, and get the function by another means. What we decide to change the logging context, that would break this. So the UDI is an essential piece of information now.

Reply:
Sure, we could do that